pkg/sql/opt/cat/column.go

```diff
@@ -45,6 +45,11 @@ type Column interface {
 	// a different ID. See the comment for StableID for more detail.
 	ColID() StableID
 
+	// PhysicalColID returns the physical column ID of a column. This ID is not
+	// stable and can be changed for example through an ALTER COLUMN TYPE command.
+	// It is guaranteed that the PhysicalColID will not be reused.
+	PhysicalColID() StableID
```

Comment:
It's confusing to use the `StableID` type for something specifically not stable. Can this be a `ColumnID` instead?

Reply:
Agreed, it really should be ColumnID